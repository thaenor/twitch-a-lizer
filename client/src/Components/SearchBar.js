import React, { useState } from 'react';
import Input from '@material-ui/core/Input';
import InputAdornment from '@material-ui/core/InputAdornment';
import SearchOutlined from '@material-ui/icons/SearchOutlined';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(theme => ({
  root: {
    width: 350,
    margin: '0 auto',
    marginBottom: '30px'
  }
}));

const SearchBar = props => {
  const classes = useStyles();
  let typingTimer;
  const typingInterval = 500;
  const [value, setValue] = useState();

  function handleChange(event) {
    setValue(event.target.value);
  }

  function doneTyping() {
    props.onSearchChange(value);
  }

  return (
    <Input
      className={classes.root}
      autoFocus
      onKeyDown={() => {
        clearTimeout(typingTimer);
      }}
      onKeyUp={() => {
        clearTimeout(typingTimer);
        typingTimer = setTimeout(doneTyping, typingInterval);
      }}
      onChange={handleChange}
      startAdornment={
        <InputAdornment position="start">
          <SearchOutlined />
        </InputAdornment>
      }
      type="text"
      placeholder="search streams"
    />
  );
};

export default SearchBar;
