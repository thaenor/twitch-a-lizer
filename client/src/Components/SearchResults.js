import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import Divider from '@material-ui/core/Divider';
import ListItemText from '@material-ui/core/ListItemText';
import ListItemAvatar from '@material-ui/core/ListItemAvatar';
import Avatar from '@material-ui/core/Avatar';
import Typography from '@material-ui/core/Typography';
import { Link } from 'react-router-dom';

const useStyles = makeStyles(theme => ({
  root: {
    width: '100%',
    maxWidth: 500,
    textAlign: 'center',
    margin: '0 auto',
    backgroundColor: theme.palette.background.paper
  },
  inline: {
    display: 'inline'
  },
  link: {
    textDecoration: 'none'
  }
}));

const SearchResults = props => {
  const classes = useStyles();

  const streamList = (
    <List className={classes.root}>
      {props.data.map(stream => {
        const linkTo = {
          pathname: `/watch/${stream.channel.name}`,
          state: stream
        };

        return (
          <div key={stream._id} id={stream._id} onClick={handleClick}>
            <Link to={linkTo} className={classes.link}>
              <ListItem alignItems="flex-start">
                <ListItemAvatar>
                  <Avatar alt={stream.game} src={stream.preview.small} />
                </ListItemAvatar>
                <ListItemText
                  primary={stream.game}
                  secondary={
                    <React.Fragment>
                      <Typography
                        component="span"
                        variant="body2"
                        className={classes.inline}
                        color="textPrimary"
                      >
                        {stream.channel.display_name}
                      </Typography>
                      <br />
                      {stream.channel.status}
                    </React.Fragment>
                  }
                />
              </ListItem>
              <Divider variant="inset" component="li" />
            </Link>
          </div>
        );
      })}
    </List>
  );

  function handleClick(event) {
    const clickedStreamId = event.currentTarget.id;
    const stream = props.data.find(element => {
      return element._id == clickedStreamId; // eslint-disable-line eqeqeq
    });
    props.handleListClick(stream);
  }

  return streamList;
};

export default SearchResults;
