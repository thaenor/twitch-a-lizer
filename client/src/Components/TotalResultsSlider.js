import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";
import Slider from "@material-ui/core/Slider";

const useStyles = makeStyles(theme => ({
  root: {
    width: 300,
    margin: "0 auto"
  },
  margin: {
    height: theme.spacing(3)
  }
}));

const TotalResultsSlider = props => {
  const classes = useStyles();

  function valuetext(value) {
    return `${value} results`;
  }

  function handleChange(event, value) {
    props.changeValue(value);
  }

  return (
    <div className={classes.root}>
      <Typography id="discrete-slider" gutterBottom>
        Total Results {props.currentValue}
      </Typography>
      <Slider
        defaultValue={props.currentValue}
        getAriaValueText={valuetext}
        aria-labelledby="discrete-slider"
        valueLabelDisplay="auto"
        onChange={handleChange}
        step={10}
        marks
        min={1}
        max={100}
      />
    </div>
  );
};

export default TotalResultsSlider;
