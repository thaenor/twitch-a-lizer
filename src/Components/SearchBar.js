import React from "react";
import Input from "@material-ui/core/Input";
import InputAdornment from "@material-ui/core/InputAdornment";
import SearchOutlined from "@material-ui/icons/SearchOutlined";
import Button from "@material-ui/core/Button";

const SearchBar = props => {
  function handleKeyDown(event) {
    if (event.keyCode === 13) {
      event.target.value !== ""
        ? props.onSearchChange(event.target.value)
        : alert("you need to search something!");
    }
  }

  return (
    <>
      <Input
        autoFocus
        onKeyDown={handleKeyDown}
        startAdornment={
          <InputAdornment position="start">
            <SearchOutlined />
          </InputAdornment>
        }
        type="text"
        placeholder="search streams"
      />
      <Button variant="contained">Search</Button>
    </>
  );
};

export default SearchBar;
