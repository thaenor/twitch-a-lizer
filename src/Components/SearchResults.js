import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import Divider from "@material-ui/core/Divider";
import ListItemText from "@material-ui/core/ListItemText";
import ListItemAvatar from "@material-ui/core/ListItemAvatar";
import Avatar from "@material-ui/core/Avatar";
import Typography from "@material-ui/core/Typography";

const useStyles = makeStyles(theme => ({
  root: {
    width: "100%",
    maxWidth: 500,
    textAlign: "center",
    margin: "0 auto",
    backgroundColor: theme.palette.background.paper
  },
  inline: {
    display: "inline"
  }
}));

const SearchResults = props => {
  const classes = useStyles();

  const streamList = (
    <List className={classes.root}>
      {props.data.streams.map(stream => {
        return (
          <div key={stream._id} id={stream._id} onClick={handleClick}>
            <ListItem alignItems="flex-start">
              <ListItemAvatar>
                <Avatar alt={stream.game} src={stream.preview.small} />
              </ListItemAvatar>
              <ListItemText
                primary={stream.game}
                secondary={
                  <React.Fragment>
                    <Typography
                      component="span"
                      variant="body2"
                      className={classes.inline}
                      color="textPrimary"
                    >
                      {stream.channel.display_name}
                    </Typography>
                    <br />
                    {stream.channel.status}
                  </React.Fragment>
                }
              />
            </ListItem>
            <Divider variant="inset" component="li" />
          </div>
        );
      })}
    </List>
  );

  function handleClick(event) {
    const clickedStreamId = event.currentTarget.id;
    const stream = props.data.streams.find(element => {
      return element._id == clickedStreamId; // eslint-disable-line eqeqeq
    });
    props.handleListClick(stream);
  }

  return streamList;
};

export default SearchResults;
