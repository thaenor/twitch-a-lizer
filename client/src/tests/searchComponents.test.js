import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import SearchBar from '../Components/SearchBar';
import SearchResults from '../Components/SearchResults';
import TotalResultsSlider from '../Components/TotalResultsSlider';
import { mock } from './fixtures/mockdata';

test('SearchBar renders correctly', () => {
  const onChangeMock = jest.fn();
  const tree = renderer
    .create(<SearchBar onSearchChange={onChangeMock} />)
    .toJSON();

  expect(tree.type).toBe('div');
  expect(tree.children).toHaveLength(2);
  expect(tree).toMatchSnapshot();
});

test('SearchResults renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <SearchResults data={mock.streams} />
      </MemoryRouter>
    )
    .toJSON();

  expect(tree.type).toBe('ul');
  expect(tree.children).toHaveLength(3);
  expect(tree).toMatchSnapshot();
});

test('TotalResultsSlider renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <TotalResultsSlider />
      </MemoryRouter>
    )
    .toJSON();

  expect(tree.type).toBe('div');
  expect(tree.children).toHaveLength(2);
  expect(tree).toMatchSnapshot();
});
