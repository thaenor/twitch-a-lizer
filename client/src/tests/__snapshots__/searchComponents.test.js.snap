// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders correctly 1`] = `
<div
  className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-adornedStart"
  onClick={[Function]}
>
  <div
    className="MuiInputAdornment-root MuiInputAdornment-positionStart"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
      />
    </svg>
  </div>
  <input
    autoFocus={true}
    className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    placeholder="search streams"
    type="text"
  />
</div>
`;

exports[`SearchResults renders correctly 1`] = `
<ul
  className="MuiList-root makeStyles-root-44 MuiList-padding"
>
  <div
    id={35077447744}
    onClick={[Function]}
  >
    <a
      className="makeStyles-link-46"
      href="/watch/sonicshadowsilver2"
      onClick={[Function]}
    >
      <li
        className="MuiListItem-root MuiListItem-gutters MuiListItem-alignItemsFlexStart"
        disabled={false}
      >
        <div
          className="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
        >
          <div
            className="MuiAvatar-root"
          >
            <img
              alt="Final Fantasy X"
              className="MuiAvatar-img"
              src="https://static-cdn.jtvnw.net/previews-ttv/live_user_sonicshadowsilver2-80x45.jpg"
            />
          </div>
        </div>
        <div
          className="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
          >
            Final Fantasy X
          </span>
          <p
            className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            <span
              className="MuiTypography-root makeStyles-inline-45 MuiTypography-body2 MuiTypography-colorTextPrimary"
            >
              Sonicshadowsilver2
            </span>
            <br />
            !mod Better FFX Mod - Any% run (Should finish this time!)
          </p>
        </div>
      </li>
      <li
        className="MuiDivider-root MuiDivider-inset"
        role="separator"
      />
    </a>
  </div>
  <div
    id={35075358048}
    onClick={[Function]}
  >
    <a
      className="makeStyles-link-46"
      href="/watch/sonicunleashed42"
      onClick={[Function]}
    >
      <li
        className="MuiListItem-root MuiListItem-gutters MuiListItem-alignItemsFlexStart"
        disabled={false}
      >
        <div
          className="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
        >
          <div
            className="MuiAvatar-root"
          >
            <img
              alt="Sonic Adventure DX: Director's Cut"
              className="MuiAvatar-img"
              src="https://static-cdn.jtvnw.net/previews-ttv/live_user_sonicunleashed42-80x45.jpg"
            />
          </div>
        </div>
        <div
          className="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
          >
            Sonic Adventure DX: Director's Cut
          </span>
          <p
            className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            <span
              className="MuiTypography-root makeStyles-inline-45 MuiTypography-body2 MuiTypography-colorTextPrimary"
            >
              SonicUnleashed42
            </span>
            <br />
            Unleasheds Adventure! - Sonic Adventure DX
          </p>
        </div>
      </li>
      <li
        className="MuiDivider-root MuiDivider-inset"
        role="separator"
      />
    </a>
  </div>
  <div
    id={35077211728}
    onClick={[Function]}
  >
    <a
      className="makeStyles-link-46"
      href="/watch/sonicsemtex"
      onClick={[Function]}
    >
      <li
        className="MuiListItem-root MuiListItem-gutters MuiListItem-alignItemsFlexStart"
        disabled={false}
      >
        <div
          className="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
        >
          <div
            className="MuiAvatar-root"
          >
            <img
              alt="Rocket League"
              className="MuiAvatar-img"
              src="https://static-cdn.jtvnw.net/previews-ttv/live_user_sonicsemtex-80x45.jpg"
            />
          </div>
        </div>
        <div
          className="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
          >
            Rocket League
          </span>
          <p
            className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            <span
              className="MuiTypography-root makeStyles-inline-45 MuiTypography-body2 MuiTypography-colorTextPrimary"
            >
              SonicSemtex
            </span>
            <br />
            PC-UK | Rocket Pleb \\\\ chill until 12:30(am) GMT
          </p>
        </div>
      </li>
      <li
        className="MuiDivider-root MuiDivider-inset"
        role="separator"
      />
    </a>
  </div>
</ul>
`;

exports[`TotalResultsSlider renders correctly 1`] = `
<div
  className="makeStyles-root-108"
>
  <p
    className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
    id="discrete-slider"
  >
    Total Results 
  </p>
  <span
    className="MuiSlider-root"
    onMouseDown={[Function]}
  >
    <span
      className="MuiSlider-rail"
    />
    <span
      className="MuiSlider-track"
      style={
        Object {
          "left": "0%",
          "width": "NaN%",
        }
      }
    />
    <input
      type="hidden"
      value=""
    />
    <span
      className="MuiSlider-mark"
      style={
        Object {
          "left": "0%",
        }
      }
    />
    <span
      className="MuiSlider-markLabel"
      style={
        Object {
          "left": "0%",
        }
      }
    />
    <span
      className="MuiSlider-mark"
      style={
        Object {
          "left": "10.1010101010101%",
        }
      }
    />
    <span
      className="MuiSlider-markLabel"
      style={
        Object {
          "left": "10.1010101010101%",
        }
      }
    />
    <span
      className="MuiSlider-mark"
      style={
        Object {
          "left": "20.2020202020202%",
        }
      }
    />
    <span
      className="MuiSlider-markLabel"
      style={
        Object {
          "left": "20.2020202020202%",
        }
      }
    />
    <span
      className="MuiSlider-mark"
      style={
        Object {
          "left": "30.303030303030305%",
        }
      }
    />
    <span
      className="MuiSlider-markLabel"
      style={
        Object {
          "left": "30.303030303030305%",
        }
      }
    />
    <span
      className="MuiSlider-mark"
      style={
        Object {
          "left": "40.4040404040404%",
        }
      }
    />
    <span
      className="MuiSlider-markLabel"
      style={
        Object {
          "left": "40.4040404040404%",
        }
      }
    />
    <span
      className="MuiSlider-mark"
      style={
        Object {
          "left": "50.505050505050505%",
        }
      }
    />
    <span
      className="MuiSlider-markLabel"
      style={
        Object {
          "left": "50.505050505050505%",
        }
      }
    />
    <span
      className="MuiSlider-mark"
      style={
        Object {
          "left": "60.60606060606061%",
        }
      }
    />
    <span
      className="MuiSlider-markLabel"
      style={
        Object {
          "left": "60.60606060606061%",
        }
      }
    />
    <span
      className="MuiSlider-mark"
      style={
        Object {
          "left": "70.70707070707071%",
        }
      }
    />
    <span
      className="MuiSlider-markLabel"
      style={
        Object {
          "left": "70.70707070707071%",
        }
      }
    />
    <span
      className="MuiSlider-mark"
      style={
        Object {
          "left": "80.8080808080808%",
        }
      }
    />
    <span
      className="MuiSlider-markLabel"
      style={
        Object {
          "left": "80.8080808080808%",
        }
      }
    />
    <span
      className="MuiSlider-mark"
      style={
        Object {
          "left": "90.9090909090909%",
        }
      }
    />
    <span
      className="MuiSlider-markLabel"
      style={
        Object {
          "left": "90.9090909090909%",
        }
      }
    />
    <span
      aria-labelledby="discrete-slider"
      aria-orientation="horizontal"
      aria-valuemax={100}
      aria-valuemin={1}
      aria-valuetext="undefined results"
      className="MuiSlider-thumb PrivateValueLabel-thumb-124"
      data-index={0}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      role="slider"
      style={
        Object {
          "left": "NaN%",
        }
      }
      tabIndex={0}
    >
      <span
        className="PrivateValueLabel-offset-126 MuiSlider-valueLabel"
      >
        <span
          className="PrivateValueLabel-circle-127"
        >
          <span
            className="PrivateValueLabel-label-128"
          />
        </span>
      </span>
    </span>
  </span>
</div>
`;
